### PR TITLE
travis build stages

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** 76d75e19c7717e954c91a4a4c4c6499aa5f7b20a

## Description

Adds [build stages](https://docs.travis-ci.com/user/build-stages/) to travis-ci to make it easier to see what failed.

image

## Contributor

- [ ] Test(s) exist to ensure functionality and minimize regresstion (if no tests added, list tests covering this PR); or,
- [x] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by joshbruce** [comment]
I really like, and am assuming here, that lint comes last; so, we get the really really important sorted first, yeah?

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-    sed -i s/0o755/0755/ test/index.js;
-    npm test;
-  else
-    npm run test:lint && npm test;
-  fi
+
+jobs:
+  include:
+    - stage: spec tests 👩🏽‍💻
+      node_js: v0.10
```
Comment (styfle) [inline .travis.yml]: It looks like this ran properly but why is the first `node_js` line missing the `-` hyphen prefix?

Comment (UziTech) [reply .travis.yml]: that is the same as json:

```json
[
      {
        "stage": "spec tests 👩🏽‍💻",
        "node_js": "v0.10"
      },
      {
        "node_js": "v4"
      }
]
```

the hyphen is the start of an array item.

**Comment by UziTech** [comment]
@joshbruce The jobs in each stage run in parallel but each stage runs in series so it will only run Lint if all Test jobs pass.

I was thinking it would be good to add stages for unit and integration tests after they are added.

We could also do a minify stage that would minify and commit after every master branch commit

**Comment by joshbruce** [comment]
> We could also do a minify stage that would minify and commit after every master branch commit

Could be a good thing. If we do that, we should make a note to update the RELEASING docs.

**Comment by joshbruce** [comment]
#1119

**Comment by UziTech** [comment]
this ☝️ should be on #1132

**Pull request merged**